modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/index/contributor/CPDefinitionModelDocumentContributor.java

```diff
@@ -508,6 +508,11 @@ public void contribute(Document document, CPDefinition cpDefinition) {
 			_expandoBridgeIndexer.addAttributes(
 				document, cpDefinition.getExpandoBridge());
 
+			for (CPInstance cpInstance : cpDefinition.getCPInstances()) {
```

Comment:
why did you put this logic here and not in the CPInstanceModelDocumentCOntributor class?
Same for the test why did you use the cpDefinitionIndexer rather than the CPInstanceIndexer?

Reply:
@andrea-ale-sbarra ,

The use-case that is being fixed by the commits here is for when search is being executed on CPDefinition Document.

We want to be able to support searching of CPDefinition depending on the custom fields of the SKU that belongs to the CPDefinition document.

Reply:
Fine then.
Thank you for the clarification!